src/reducers/timeline.js

```diff
@@ -0,0 +1,10 @@
+import { Map } from 'immutable';
+
+export default ( state = new Map(), action ) => {
```

Comment:
I would rename the reducers, actions, and epics to be `revisions`. Since the timeline is going to be filtered revisions, then the timeline is an implementation of the revisions.